src/conversion/buildviewconverter.js

```diff
@@ -141,11 +141,21 @@ class ViewConverterBuilder {
 	 * @param {String|RegExp} [value] View attribute value.
 	 * @returns {module:engine/conversion/buildviewconverter~ViewConverterBuilder}
 	 */
-	fromAttribute( key, value = /.*/ ) {
+	fromAttribute( key, value = /.+/ ) {
 		let pattern = {};
-		pattern[ key ] = value;
 
-		return this.from( pattern );
+		if ( key === 'style' || key === 'class' ) {
+			pattern[ key ] = value;
+		} else {
+			pattern.attribute = {};
+			pattern.attribute[ key ] = value;
+		}
+
+		this.from( pattern );
+
+		this._from[ this._from.length - 1 ].attributeKey = key;
```

Comment:
You pass it to `from()` but you can also use `from()` directly and pass whatever matcher/match object you want, so it is impossible to sniff whether `.fromAttribute( x ).toAttribute( y )` was used or not. I could check in `.toAttribute()` if `from` has just one value and it is `attribute` which has just one value, but I feel that that would be even uglier :P